superset/views/database/forms.py

```diff
@@ -207,3 +207,177 @@ def at_least_one_schema_is_allowed(database):
+        excel_allowed_dbs = []
+        # TODO: change allow_csv_upload to allow_file_upload
+        excel_enabled_dbs = (
+            db.session.query(models.Database).filter_by(allow_csv_upload=True).all()
+        )
+        for excel_enabled_db in excel_enabled_dbs:
+            if CsvToDatabaseForm.at_least_one_schema_is_allowed(excel_enabled_db):
+                excel_allowed_dbs.append(excel_enabled_db)
+        return excel_allowed_dbs
+
+    @staticmethod
+    def at_least_one_schema_is_allowed(database):
+        """
+        If the user has access to the database or all datasource
+            1. if schemas_allowed_for_csv_upload is empty
+                a) if database does not support schema
+                    user is able to upload csv without specifying schema name
+                b) if database supports schema
+                    user is able to upload csv to any schema
+            2. if schemas_allowed_for_csv_upload is not empty
+                a) if database does not support schema
+                    This situation is impossible and upload will fail
+                b) if database supports schema
+                    user is able to upload to schema in schemas_allowed_for_csv_upload
+        elif the user does not access to the database or all datasource
+            1. if schemas_allowed_for_csv_upload is empty
+                a) if database does not support schema
+                    user is unable to upload csv
+                b) if database supports schema
+                    user is unable to upload csv
+            2. if schemas_allowed_for_csv_upload is not empty
+                a) if database does not support schema
+                    This situation is impossible and user is unable to upload csv
+                b) if database supports schema
+                    user is able to upload to schema in schemas_allowed_for_csv_upload
+        """
+        if (
+            security_manager.database_access(database)
+            or security_manager.all_datasource_access()
+        ):
+            return True
+        schemas = database.get_schema_access_for_csv_upload()
+        if schemas and security_manager.schemas_accessible_by_user(
+            database, schemas, False
+        ):
+            return True
+        return False
+
+    name = StringField(
+        _("Table Name"),
+        description=_("Name of table to be created from excel data."),
+        validators=[DataRequired()],
+        widget=BS3TextFieldWidget(),
+    )
+    excel_file = FileField(
+        _("Excel File"),
+        description=_("Select a Excel file to be uploaded to a database."),
+        validators=[
+            FileRequired(),
+            FileAllowed(
+                config["ALLOWED_EXTENSIONS"],
+                _(
+                    "Only the following file extensions are allowed: "
+                    "%(allowed_extensions)s",
+                    allowed_extensions=", ".join(config["ALLOWED_EXTENSIONS"]),
```

Comment:
Here we should probably only check for allowed Excel extensions, i.e. the intersection of `ALLOWED_EXTENSIONS` and `EXCEL_EXTENSION`